### PR TITLE
Add dnsmasq configuration per BZ

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1518022

### CODE REVIEW

**Comment by mburke5678** [comment]
@rjhowe PTAL. I took the 6 steps of your [Comment 1](https://bugzilla.redhat.com/show_bug.cgi?id=1518022#c1) and brought them into the manual upgrade topic. Please let me know if this is what is needed.

**Comment by mburke5678** [comment]
@rjhowe Post shutdown BUMP! ^^

**Comment by DanyC97** [comment]
@mburke5678 isn't the steps in your PR covered by the advance installation (openshift-ansible) method ?

**Comment by rjhowe** [comment]
If this is going in orgin docs this will work. For enterprise docs we will need to direct the user to pull this script from the openshift playbooks rpm .

# wget https://raw.githubusercontent.com/openshift/openshift-ansible/release-3.6/roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh -O /etc/NetworkManager/dispatcher.d/99-origin-dns.sh

**Comment by mburke5678** [comment]
@rjhowe I a not clear on which steps are needed for Enterprise vs. Online. 

The step to download the script is in the file now; that is for Enterprise only? 

Does the user need to be concerned with the steps in the script:
_To use this,
 - If this host is also a master, reconfigure master dnsConfig to listen on
   8053 to avoid conflicts on port 53 and open port 8053 in the firewall
 - Drop this script in /etc/NetworkManager/dispatcher.d/systemctl restart NetworkManager
 - Configure node-config.yaml to set dnsIP: to the ip address of this node_

WIP Enterprise:
http://file.rdu.redhat.com/mburke/BZ-1518022/install_config/upgrading/manual_upgrades.html#preparing-for-a-manual-upgrade  (Step 4)

WIP Origin:
http://file.rdu.redhat.com/mburke/openshift-origin/BZ-1518022/BZ-1518022/install_config/upgrading/manual_upgrades.html#upgrading-masters (Step 5)

**Comment by mburke5678** [comment]
@rjhowe PTAL

**Comment by mburke5678** [comment]
@rjhowe BUMP

**Comment by mburke5678** [comment]
@rjhowe I was asked by @vikram-redhat to complete this BZ. Can you help with my questions above?

**Comment by mburke5678** [comment]
@wmengRH PTAL

**Comment by mburke5678** [comment]
@wmengRH BUMP

File: install_config/upgrading/manual_upgrades.adoc [hunk 1]
```diff
@@ -269,6 +269,89 @@ For multi-master clusters:
...
+enable-dbus
+bind-interfaces
+listen-address=<node_ip_address> 
+----
+
+.. Edit the *_/etc/dnsmasq.d/origin-upstream-dns.conf_* file as follows:
++
+---- 
+server=<dns_server1_ip_address>
+server=<dns_server1_ip_address>
```
Comment (wmengRH) [inline install_config/upgrading/manual_upgrades.adoc]: server2

**Comment by wmengRH** [comment]
LGTM.
Thanks.

**Comment by mburke5678** [comment]
@rjhowe @wmengRH @sjwalter2 
In these conf files, what are the node-ip-addresses?
```
origin-dns.conf 
listen-address=<NODE IP ADDRESS HERE> 

```
The Docker registry service IP address.?

```
node-config.yaml 
dnsIP: <NODE IP ADDRESS> 

```
IP address of the DNS name server??

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

Added new step to Upgrading Master Components:
Step 4 in OCP:
http://file.rdu.redhat.com/mburke/BZ-1518022/install_config/upgrading/manual_upgrades.html#upgrading-masters
Step 5 in Origin:
http://file.rdu.redhat.com/mburke/origin/install_config/upgrading/manual_upgrades.html#upgrading-masters

File: install_config/upgrading/manual_upgrades.adoc [hunk 2]
```diff
@@ -269,6 +269,93 @@ For multi-master clusters:
 # systemctl enable origin-master-controllers
 ----
 
+. Configure Dnsmasq. Because of changes made in {product-title} 3.6, you will need to 
+perform the following steps to configure dnsmasq as part of the upgrade. See DNS Changes 
+under xref:../../release_notes/ocp_3_6_release_notes.adoc#ocp-36-notable-technical-changes[Notable Technical Changes]
+in the {product-title} 3.6 Release Notes.
++
+.. Create a *_/etc/origin/node/node-dnsmasq.conf_*  node configuration file
```
Comment (tmorriso-rh) [inline install_config/upgrading/manual_upgrades.adoc]: Add a period after file.

**Comment by tmorriso-rh** [comment]
One comment. Otherwise, LGTM.

**Comment by wmengRH** [comment]
@mburke5678 
"NODE IP ADDRESS" is not DNS server IP address, 
is IP address of the host (node located), usually IP address of eth0, which means it varies each node.
on host, execute command
 ip address 
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether fa:16:3e:87:44:c3 brd ff:ff:ff:ff:ff:ff
    inet 172.16.120.10/24 brd 172.16.120.255 scope global dynamic eth0

"NODE IP ADDRESS" is 172.16.120.10 on this node
master is also a node

**Comment by mburke5678** [comment]
@rjhowe @wmengRH These steps are needed only for upgrading from 3.5 to 3.6? Once updating to 3.6, the problem is taken care of?

**Comment by wmengRH** [comment]
I think so. 
I fresh installed OCP 3.5 and 3.6, the configs mentioned in this PR which are not in 3.5,  are in OCP 3.6 by default

**Pull request merged**